### ISSUE TITLE
Question about few-shot learning of MRE

### ISSUE THREAD

**Issue by stepjames1998** [zjunlp/MKGformer#7]
Hi there, nice work. There is a detail I am not very clear.  In few-shot learning of MRE, did you just randomly sample K examples for each relation type to construct the few-shot training dataset and train MKGformer using the constructed dataset？ I noticed that in FL-MSRE, some few-shot techniques are used, did you use similar techniques in this paper?

**Comment by stepjames1998** [comment]
Many thanks!

**Comment by stepjames1998** [comment]
We noticed that the selection of None type ratio affects the experimental results in the experiments of new-shot learning. I wonder how the ratio of None type is set in your paper when constructing small sample datasets? @zxlzr @Flow3rDown

**Comment by flow3rdown** [comment]
Thanks for your attention! 

We follow the low-resource setting of [1], which is different from the N-way K- shot setting. Specifically, we randomly sample k instances of each relation using five different seeds and average the results.

[1] Making Pre-trained Language Models Better Few-shot Learners. ACL/IJCNLP 2021.

**Comment by stepjames1998** [comment]
> 

Well, the paper you cited does not consider the relation extraction task which has None type examples, do you mean that you sample K examples even when the relation type is None? 
We find that in MNRE dataset, there are too many None relations in the dataset, and average sampling may be biased. Actually,  under the average setting, the reproduced metrics are significantly lower than the paper. We also test sampling more None type examples, and the results improve immediately.

**Comment by njcx-ai** [comment]
Yes, we sample K examples in the k-shot setting even when the relation type is None. The problem of reproduced metrics may be due to the randomness of sampling.

**Comment by Luyuxi00** [comment]
How do you set the batch size, learning rate in the few-shot learning stage?
